lib/host-manager.js

```diff
@@ -6,32 +6,52 @@ const Table = require('cli-table')
 const Config = require('../lib/config.js')
 const esh = require('eshost');
 
+const head = [
+  'name',
+  'type',
+  'path',
+  'args',
+  'tags',
+];
+
+const colWidths = Array(head.length).fill(0);
+
 exports.list = function list(config) {
   console.log(chalk.grey('Using config ', config.configPath));
 
-  var table = new Table({
-    head: [
-      'name',
-      'type',
-      'path',
-      'args',
-      'tags'
-    ],
+  const table = new Table({
+    colWidths, head
   });
 
-  Object.keys(config.hosts).forEach(name => {
-    const host = config.hosts[name];
-    table.push([
-      name,
-      host.type,
-      host.path || '',
-      host.args || '',
-      (host.tags || '') + '',
-    ]);
-  })
-
-  console.log(table.toString());
-}
+  const names = Object.keys(config.hosts);
+  let output;
+
+  if (!names.length) {
+    output = 'No configured hosts';
+  } else {
+    output = names.reduce((accum, name) => {
+      const {
+        type,
+        path = '',
+        args = '',
+        tags = ''
+      } = config.hosts[name];
+
+      const row = [ name, type, path, args, tags ];
+      const {colWidths: widths} = accum.options;
+
+      // Update the stored colWidths for each cell, saving the largest/longest
+      accum.options.colWidths = Array.from(widths, (width, index) => {
+        return Math.max(width, String(row[index]).length) + 2;
+      });
+
+      accum.push(row);
+      return accum;
+    }, table).toString();
```

Comment:
Ah I see you're going with the method of manually calculating the column widths. I was hoping to avoid that since it seems the purview of the library, but since the work is already done, **HUZZAH!!!**